tests/unit/components/pilas-blockly-test.js

```diff
@@ -215,5 +215,145 @@ module('Unit | Components | pilas-blockly', function (hooks) {
     assertHasProps(assert, metadata, 'ast', 'staticAnalysis', 'turboModeOn',)
     assert.ok(metadata.program || metadata.program.length === 0)
   })
+})
+
+module('Unit | Components | pilas-blockly | ToolboxForBlockTypes', function(hooks) {
+  setupTest(hooks)
+
+  const operator = { categoryId: 'operators' }
+  const primitive = { categoryId: 'primitives' }
+
+  const separator = { categoryId: 'separator', isSeparator: true }
+  const blockId = "A block id"
+
+  const doThis = 'Do this'
+  const doThat = 'Do that'
+
+  const myProcedures = { categoryId: 'procedures', blocks: [doThis, doThat] }
+
+  const toolbox = [blockId, myProcedures, separator]
+
+  const emberBlocklySeparator = {
+    isSeparator: true
+  }
+
+  hooks.beforeEach(function () {
+    this.owner.lookup('service:blocksGallery').start()
+    this.ctrl = this.owner.factoryFor('component:pilas-blockly').create()
+    this.ctrl.set('modelActividad', actividadMock)
+    this.ctrl.modelActividad.set('estiloToolbox', 'conCategorias')
+    this.ctrl.get('intl').setLocale('en-us') // This is necessary because categories are tied to locale and translation.
+  })
+
+  function setCategoriesNotRequired(ctrl) {
+    ctrl.modelActividad.set('estiloToolbox', 'sinCategorias')
+  }
+
+  test('ToolboxForBlockTypes should fail if blockTypes is undefined', function (assert) {
+    assert.throws(() => this.ctrl.toolboxForBlockTypes(undefined))
+  })
+
+  test('Toolbox should be sorted based on Pilas Bloques stablished order', function (assert) {
+    const alternative = { categoryId: 'alternatives' }
+    assert.propEqual(this.ctrl.sortedToolbox([operator, primitive, alternative]), [primitive, alternative, operator])
+  })
+
+  test('When sorted categories that are not acknowledged by Pilas Bloques should be at the end', function (assert) {
```

Comment:
```suggestion
  test('Uncategorized category should be the last item in the toolbox after sorting', function (assert) {
```

Acá estás probando que si hay categoría uncategorized se ponga al final. No estás probando que los bloques sin categoría queden dentro de esa, eso lo probaste abajo ya.